reyna/src/com/b2msolutions/reyna/Repository.java

```diff
@@ -12,144 +12,270 @@
 
 public class Repository extends SQLiteOpenHelper {
 
-	public static final String DATABASE_NAME = "reyna.db";
+    public static final String DATABASE_NAME = "reyna.db";
 
-	public static final int DATABASE_VERSION = 1;
-	
-	private static final String TAG = "Repository";
+    public static final int DATABASE_VERSION = 1;
 
-	public Repository(Context context) {
-		super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    private static final String TAG = "Repository";
+
+    private static final int SIZE_DIFFERENCE_TO_START_CLEANING = 307200; //300Kb in bytes
+
+    private static final int SIZE_DIFFERENCE_TO_START_VACUUM = 1048576; //1Mb in bytes
```

Comment:
One MB feels slightly arbitrary, why not simply check if db over limit in the function that sets size limit, and trigger vacuuming from there.